src/plugins/data/server/ui_settings.ts

```diff
@@ -535,6 +535,17 @@ export function getUiSettings(
       }),
       schema: schema.string(),
     },
+    [UI_SETTINGS.DATA_WITH_LONG_NUMERALS]: {
+      name: i18n.translate('data.advancedSettings.data.withLongNumeralsTitle', {
+        defaultMessage: 'Extend Numeric Precision',
+      }),
+      value: true,
+      description: i18n.translate('data.advancedSettings.data.withLongNumeralsText', {
+        defaultMessage:
+          "Turn on for precise handling of extremely large numbers. Turn off to optimize performance when high precision for large values isn't required.",
```

Comment:
should this tell user what the threshold is before losing precision?

Reply:
It would be too wordy. Instead, I will have this added to the documentation website with proper explanation.